### ISSUE TITLE
Expose Lighthouse recommendations (including WP-specific) in PageSpeed widget

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2390]
The `DashboardPageSpeedWidget` should be expanded to surface optimization recommendations from the API. These recommendations are already included in the API responses the plugin currently receives, so this issue is purely focused on implementing new components and UI for it.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The PageSpeed widget should receive an additional section below the current main content, separated by a gray line, which holds an accordion with all applicable recommendations from the API response.
    * The UI for the recommendations should look like in the [Figma file](https://www.figma.com/file/niZV5sVfeaE5sqnJAUgR3G/PageSpeed-Insights?node-id=160%3A37). (Note that the Figma mock also has a modified refresh link - this shouldn't be changed as part of this issue though as it's covered separately by #2391.)
    * The contents of the accordion should be based on the current device (the API response for `mobile` _may_ return different values and recommendations than the one for `desktop`).
    * The recommendations to display should be determined as follows:
        * Look at `lighthouseResult.audits` in the API response.
        * Only display recommendations that satisfy all of the following criteria:
            * The `scoreDisplayMode` is set to `numeric` or `binary`.
            * The `score` is lower than 0.9.
            * There is a WordPress stack pack description available for the recommendation (see `lighthouseResult.stackPacks` array).
        * Sort the recommendations so that those with lower scores (i.e. the most critical recommendations) come first.
        * For the title of each accordion section, use the audit's `title`.
        * For the content of each accordion section, use the WordPress stack pack description. The "general" description and any details about the recommendations should **not** be displayed for now.
        * Any of these descriptions may contain links, in Markdown format. So the descriptions need to be parsed to transform those Markdown links into HTML `a` tags. All such links should open in a new window via `target="blank"`.
* This change should apply to both the current PSI widget and the new PSI widget using the widgets API (likely trivial because they already reuse the same components for the most part).

#### New selectors and utility function
* Create a new utility function `markdownToHtml` in the `assets/js/util/markdown.js` file. It should replace markdown links (`[...](...)`) in the text into actual HTML `a` tags with `target=_blank` attribute and appropriate `noopener noreferrer` rel attribute.
* Edit the `assets/js/modules/pagespeed-insights/datastore/report.js` file:
  * Add a new selector `getAudits` that should accept `url` and `strategy` arguments, and return the `audits` list from the `lighthouseResult` without modifications;
  * Add a new selector `getStackPackDescriptions` which should accept `url`, `strategy` and `auditId` and return an array of objects composed from `stackPacks` details. Each stackPack details object in the array should consist of:
    * `id` taken from the `id` property;
    * `icon` taken from the `iconDataURL` property;
    * `title` taken from the `title` property;
    * `description` taken from the `descriptions` object using `auditId` as the key.

#### Accordion components
* Create Accordion and AccordionItem components that implement appropriate UI elements designed in the Figma file (see in the AC);
* Investigate the possibility to re-use styles defined in the `assets/sass/components/global/_googlesitekit-accordion.scss` file for the new components.
  * If it turns out that those styles aren't needed for the new accordion components, remove it since we don't use it anywhere.
* Add a new story for accordion components in the global section of the storybook.

#### Recommendation components
* Create a new `Recommendation` component in the `assets/js/modules/pagespeed-insights/components/common` folder. It should do the following:
  * Accept an audit object via props;
  * Select stackPacks using `getStackPackDescriptions` selector;
  * Render an accordion item using the title from the audit object + stackPack descriptions.
* Create a new `Recommendations` components in the `assets/js/modules/pagespeed-insights/components/common` folder. It should do the following:
  * Select audits using the new `getAudits` selector.
  * Filter out audits which `scoreDisplayMode` is neither `number` nor `binary`, `score` is >= `0.9`, or there is no WordPress related description (where `id` === `"wordpress"`) for it in the stack pack.
  * Sort remaining audits by score ascendingly, so audits with the lowest score will be at the beginning of the list.
  * Render an Accordion with audits mapped to the `Recommendation` component.
* Add the `Recommendations` component to the `DashboardPageSpeed` component. It should go at the bottom after the section with `LabReportMetrics` and `FieldReportMetrics` components.
* Make remaining UI adjustments to make sure the new recommendations have styles that match design in the Figma file.

### Test Coverage

* Add js tests for the `markdownToHtml` function;
* Add js tests for the new selectors;

### Visual Regression Changes

* The new recommendations section should appear both on the new widget and the legacy one.

## QA Brief

* Set up Site Kit on your site;
* Activate PSI module;
* Go to the dashboard and check the PSI widget;
* Make sure you see recommendations on both the new widget and the legacy one.

## Changelog entry

* Surface WordPress-specific user experience recommendations in the PageSpeed Insights widget.

**Comment by felixarntz** [comment]
IB ✅

**Comment by felixarntz** [comment]
@eugene-manuilov I've updated the ACs here based on our conversation this morning, so this is ready for the IB to be slightly adjusted to cover for the change.

**Comment by eugene-manuilov** [comment]
Thanks, @felixarntz! IB is updated.

**Comment by felixarntz** [comment]
IB ✅

**Comment by eugene-manuilov** [comment]
@aaemnnosttv @felixarntz @tofumatt this ticket is ready for the initial code review even though github shows failed checks which seem to be caused by github itsefl.

**Comment by wpdarren** [comment]
### QA Update: 

@eugene-manuilov the styling seems off on the new section, there is no margin. Also in the AC it says that there should be a gray line. Unfortunately, I do not have access to the Figma UI mock, but Is the styling being picked up in another ticket? - [Screenshot](https://user-images.githubusercontent.com/73545194/101021306-bad27800-35aa-11eb-9acb-2c9d2f0db57b.png)

**Comment by aaemnnosttv** [comment]
## QA ❌ 

The report details link is duplicated on the field data tab – the design shows this only once at the bottom.

![image](https://user-images.githubusercontent.com/1621608/101077491-93c97580-35ad-11eb-8883-5a270093e184.png)

Also, not sure if we should show recommendations when field data is not available?

![image](https://user-images.githubusercontent.com/1621608/101077652-cd01e580-35ad-11eb-8c6d-6d37b352b852.png)

cc: @felixarntz @eugene-manuilov

**Comment by felixarntz** [comment]
@aaemnnosttv Oh right, I missed that during code review. The `ReportDetailsLink` wasn't removed from `FieldReportMetrics`.

Regarding the recommendations surfacing for no field data, that is good as is and should be kept.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv @felixarntz let me fix it really quickly.

**Comment by aaemnnosttv** [comment]
@felixarntz CR ✅  – just waiting on the one E2E job to re-run – assigned back to you to merge.

**Comment by marrrmarrr** [comment]
LGTM
